test/unit/MailHelperTest.php

```diff
@@ -133,4 +134,63 @@ public function testJsonSerializeOverPersonalizationsShouldNotReturnNull()
 
         $this->assertEquals($expectedJson, $json);
     }
+
+    /**
+     * @throws TypeException
+     */
+    public function testMailPersonalizations()
+    {
+        $objEmail = new Mail('me@example.com');
+
+        // Add the first personalization.
+        $objEmail->addTo('foo@bar.com', 'foo bar', ['this' => 'that']);
+        // Update the last personalization.
+        $objEmail->setSendAt(1234567890);
+        // Update the first personalization.
+        $objEmail->addHeader('Head', 'der', 0);
+        // Append an existing personalization.
+        $objEmail->addSubstitution('sub', 'this', null, new Personalization());
+
+        $json = json_encode($objEmail, JSON_PRETTY_PRINT);
+
+        $expectedJson = <<<JSON
```

Comment:
That's cool. I didn't know you could do that.

Reply:
/me copied it from above